pkg/operator/minio/controller.go

```diff
@@ -352,7 +352,10 @@ func (c *Controller) onUpdate(oldObj, newObj interface{}) {
 }
 
 func (c *Controller) onDelete(obj interface{}) {
-	objectstore := obj.(*miniov1alpha1.ObjectStore).DeepCopy()
+	objectstore, ok := obj.(*miniov1alpha1.ObjectStore)
+	if ok {
+		objectstore = objectstore.DeepCopy()
+	}
 	logger.Infof("Delete Minio object store %s", objectstore.Name)
```

Comment:
This also uses the invalid `objectstore` if the type assertion failed

Reply:
fixed